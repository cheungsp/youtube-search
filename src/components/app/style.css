body {
  background-color: whitesmoke;
}

.video-item img {
  max-width: 64px;
}

.video-detail .details {
  margin-top: 10px;
  padding: 10px;
  border: 1px solid #ddd;
  border-radius: 4px;
}