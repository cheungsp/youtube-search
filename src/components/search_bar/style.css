.search-bar {
  margin: 20px;
  text-align: center;
}

.search-bar input {
  width: 80%;
}

/*
"test": "jest"
"test": "react-scripts test --env=jsdom",
*/