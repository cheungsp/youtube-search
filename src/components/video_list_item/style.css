.single-video-item {
  margin-bottom: 15px;
}