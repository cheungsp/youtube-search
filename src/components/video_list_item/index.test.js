import React from 'react';
import { shallow } from 'enzyme'
import VideoList from '../video_list/index.js'
import VideoListItem from './index.js'
import toJson from 'enzyme-to-json'
import './style.css';


test('VideoListItem component should render as expected given a video', () => {
  const video = {
    'kind': "youtube#searchResult",
    'etag': "VPWTmrH7dFmi4s1RqrK4tLejnRI/j0uEstXCXOhrDqDegEBmEeHqsBM",
    'snippet': {
      'thumbnails': {
        'default': {
          'url':"https://i.ytimg.com/vi/YQHsXMglC9A/default.jpg"
        }
      }
    },
    'title': 'video 1',
    'key': 1,
    'id': 1,
  }
  const component = shallow(<VideoListItem video={video}/>)  
  const tree =  toJson(component)
  expect(component.find(VideoListItem).length).toBe(0);
  expect(tree).toMatchSnapshot()
})

 