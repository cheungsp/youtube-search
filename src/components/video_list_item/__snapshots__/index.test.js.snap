// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoListItem component should render as expected given a video 1`] = `
<li
  className="list-group-item"
  onClick={[Function]}
>
  <div
    className="video-list media"
  >
    <div
      className="media-left"
    >
      <img
        className="media-object"
        src="https://i.ytimg.com/vi/YQHsXMglC9A/default.jpg"
      />
    </div>
    <div
      className="media-body"
    >
      <div
        className="media-heading"
      />
    </div>
  </div>
</li>
`;
